src/rgw/rgw_d4n_datacache.cc

```diff
@@ -0,0 +1,282 @@
+#include "rgw_d4n_datacache.h"
+#include "rgw_sal_d4n.h"
+#include <iostream>
+
+#define dout_subsys ceph_subsys_rgw
+#define dout_context g_ceph_context
+
+std::vector< std::pair<std::string, std::string> > RGWD4NCache::buildObject(rgw::sal::Attrs* baseBinary, rgw::sal::Attrs* newBinary) {
+  std::vector< std::pair<std::string, std::string> > values;
+  rgw::sal::Attrs::iterator attrs;
+ 
+  // Convert to vector
+  if (baseBinary != NULL && newBinary != NULL) {
+    for (attrs = baseBinary->begin(); attrs != baseBinary->end(); ++attrs) {
+      values.push_back(std::make_pair(attrs->first, attrs->second.to_str()));
```

Comment:
It might be worth finding a way to stream data to redis rather than convert the bufferlists to strings.